### PR TITLE
Fix sharing bug and refactor

### PR DESCRIPTION
**Pull request by notthatmj**

### CODE REVIEW

File: Bernard/FavoritesViewController.swift [hunk 1]
```diff
@@ -19,73 +21,77 @@ class FavoritesNavigationBarDelegate: NSObject, UINavigationBarDelegate {
...
+        
+        let confirmationController = UIAlertController(title: title,
+                                                            message: message,
+                                                            preferredStyle: UIAlertControllerStyle.actionSheet)
+        let okAction = UIAlertAction(title: "OK", style: UIAlertActionStyle.destructive) {(action: UIAlertAction) in
+            self.controller.clearFavorites()
+        }
+        confirmationController.addAction(okAction)
+        let cancelAction = UIAlertAction(title: "Cancel", style: UIAlertActionStyle.cancel) {(action: UIAlertAction) in return
+        }
```
Comment (jacobvanorder) [inline Bernard/FavoritesViewController.swift]: This is empty. You could just pass `.none` to that argument.

Comment (jacobvanorder) [reply Bernard/FavoritesViewController.swift]: Or `nil`

File: Bernard/FavoritesViewController.swift [hunk 2]
```diff
@@ -19,73 +21,77 @@ class FavoritesNavigationBarDelegate: NSObject, UINavigationBarDelegate {
...
-            return namesModelFavoritesCount
-        }
-        return 0
+        return self.controller.favorites.count
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
-        cell.textLabel?.text = namesModel?.favorites[indexPath.row]
+        cell.textLabel?.text = controller.favorites[indexPath.row];
```
Comment (jacobvanorder) [inline Bernard/FavoritesViewController.swift]: Any need for the semi-colon? Old habit?

File: Bernard/FavoritesViewController.swift (hunk 1 above)
Comment (notthatmj) [reply Bernard/FavoritesViewController.swift]: Good point! I removed it.

**Pull request merged**